MaraBot/Core/Option.cs

```diff
@@ -0,0 +1,84 @@
+using System.ComponentModel.DataAnnotations;
+using System.Collections.Generic;
+
+namespace MaraBot.Core
+{
+	/**
+	 * Randomizer modes the options are categorized by.
+	 */
+	public enum Mode
+	{
+		Mode,
+		General,
+		Rando,
+		Open,
+		AncientCave,
+		BossRush,
+		Chaos,
+		Other
+	}
+
+	/**
+	 * Information about a randomizer option.
+	 */
+	public struct Option
+	{
+		/**
+		 * Name of the options, as seen in the randomizer.
+		 */
+		public readonly string Name;
+
+		/**
+		 * Mode it belongs in.
+		 */
+		public readonly Mode Mode;
+
+		/**
+		 * Possible values the option can have in the options JSON/string,
+		 * together with the display name as seen in the randomizer.
+		 * Default values will not be in this dictionary,
+		 * because those values don't appear in the options string.
+		 */
+		public readonly IDictionary<string, string> Values;
+
+		public Option( string name, Mode mode, IDictionary<string, string> values )
+		{
+			Name   = name  ;
+			Mode   = mode  ;
+			Values = values;
+		}
+
+		/**
+		 * Contains a deserialization of the options config file,
+		 * containing all randomizer options.
+		 */
+		public static Dictionary<string, Option> Options;
```

Comment:
This shouldn't live inside the `Option` struct. Let's have this in `Program.cs` instead with the other deserialized data.